### PR TITLE
test: add exit_handler, loop_parallism and output_a_directory pipelines to v2 sample test

### PR DESCRIPTION
**Pull request by Bobgy**
**Description of your changes:**
Either the old sample and updated sample fails, not sure if it's related to exit handler or not.

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?

    [Learn more about cherry-picking updates into the release branch](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#cherry-picking-pull-requests-to-release-branch).

### CODE REVIEW

File: samples/core/exit_handler/exit_handler.py [hunk 1]
```diff
@@ -13,37 +13,35 @@
...
-        image='google/cloud-sdk:279.0.0',
-        command=['sh', '-c'],
-        arguments=['gsutil cat $0 | tee $1', url, '/tmp/results.txt'],
-        file_outputs={
-            'data': '/tmp/results.txt',
-        }
+import os
+from kfp import dsl, components
+
+gcs_download_op = kfp.components.load_component_from_file(
```
Comment (Ark-kun) [inline samples/core/exit_handler/exit_handler.py:20]: We could probably use `ComponentStore` configured with both local path and remote URL.

File: samples/core/output_a_directory/output_a_directory_test.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# Copyright 2021 The Kubeflow Authors
```
Comment (Ark-kun) [inline samples/core/output_a_directory/output_a_directory_test.py:1]: I wonder whether it's possible to not need these test launchers for every sample.

Perhaps we could move all of them to the sample test dir.

Comment (Bobgy) [reply samples/core/output_a_directory/output_a_directory_test.py:1]: Can you provide a sample?
I just realized local import might not be possible in legacy e2e tests https://oss-prow.knative.dev/view/gs/oss-prow/pr-logs/pull/kubeflow_pipelines/5473/kubeflow-pipeline-e2e-test/1382342847985881088#1:build-log.txt%3A5624.

I need to use an url import here.

File: samples/core/exit_handler/exit_handler.py (hunk 1 above)
Comment (Bobgy) [reply samples/core/exit_handler/exit_handler.py]: Can you provide a sample?
I just realized local import might not be possible in legacy e2e tests https://oss-prow.knative.dev/view/gs/oss-prow/pr-logs/pull/kubeflow_pipelines/5473/kubeflow-pipeline-e2e-test/1382342847985881088#1:build-log.txt%3A5624.

I need to use an url import here.

File: samples/core/output_a_directory/output_a_directory_test.py (hunk 2 above)
Comment (Bobgy) [reply samples/core/output_a_directory/output_a_directory_test.py:1]: I'd suggest keeping them here, they are useful for several reasons:

* allow custom verification logic for each test -- especially verifying the MLMD data structure (I plan to add some examples, and I believe they are the true value for a pipeline test, verifying a pipeline can succeed is not verifying much)
* allow arbitrary number of testcases with different setup (test v1_legacy, test v2_compatible, test different arguments)
* keep test next to code

**Comment by Bobgy** [comment]
/assign @neuromage @chensun @Ark-kun 
Any more comments?

File: samples/core/output_a_directory/output_a_directory_test.py (hunk 2 above)
Comment (Ark-kun) [reply samples/core/output_a_directory/output_a_directory_test.py:1]: I think we might want to have some separation between what we show to the users and the internal test infra machinery.

It's OK to have multiple test cases for a single samples, but I'm not sure we should show the users with that test infra code.

File: samples/test/config.yaml [hunk 3]
```diff
@@ -22,39 +22,47 @@
...
+  path: samples.core.output_a_directory.output_a_directory_test
 - name: loop_output
   path: samples.core.loop_output.loop_output_test
 - name: loop_static
   path: samples.core.loop_static.loop_static_test
 - name: loop_parameter
   path: samples.core.loop_parameter.loop_parameter_test
-- name: parameter_with_format
-  path: samples.test.parameter_with_format_test
+- name: loop_parallelism
```
Comment (Ark-kun) [inline samples/test/config.yaml:39]: D you think we can include the list of supported modes here as well? This way there is no need to have separate test py files alongside the tests.

In the future we could also support pipeline argument sets here.

**Comment by Bobgy** [comment]
/approve
I think remaining comments are more general than this PR just trying to add a few more samples, so I'm going to merge it and create separate issues to discuss the remaining samples.

File: samples/core/output_a_directory/output_a_directory_test.py (hunk 2 above)
Comment (Bobgy) [reply samples/core/output_a_directory/output_a_directory_test.py:1]: Created https://github.com/kubeflow/pipelines/issues/5506 to track further discussion.

File: samples/core/exit_handler/exit_handler.py (hunk 1 above)
Comment (Ark-kun) [reply samples/core/exit_handler/exit_handler.py]:
```
component_store = components.ComponentStore(
    local_search_paths=[
        '.',
    ],
    url_search_prefixes=[
        'https://raw.githubusercontent.com/kubeflow/pipelines/master/components/'
    ],
)
...
op = component_store.load_component(name='google-cloud/storage/download_blob')
```

**Pull request merged**